types/lazy/lazy.go

```diff
@@ -154,3 +154,32 @@ func SyncFuncErr[T any](fill func() (T, error)) func() (T, error) {
 		return v, err
 	}
 }
+
+// TB is a subset of testing.TB that we use to set up test helpers.
+// It's defined here to avoid pulling in the testing package.
+type TB interface {
+	Helper()
+	Cleanup(func())
+}
+
+// SetForTest sets z's value and error.
+// It's used in tests only and reverts z's state back when tb and all its
+// subtests complete.
```

Comment:
Document when it's safe to call this? At the very least, it's not safe to call `SetForTest` concurrently with another `SetForTest`, since they'll race on reading/writing `z.v`. It's probably also not safe to call `SetForTest` concurrently with another `Get`/`GetErr`, since that can probably race. In the example below, (1) and (2) are two goroutines:

```
(1) SetForTest: z.once.Do(func() {})
(2) GetErr: z.once.Do(func() { /* ... */ })
(2) GetErr: return z.v, *z.err.Load()
(1) SetForTest: z.v = val
```

Which would result in `GetErr` returning the zero value and nil error, even if `SetForTest` is called with something else.

Reply:
Good point, thank you. I assumed a common use pattern where the value is set at the very beginning of the test before it (or the code being tested) creates any goroutines, but we should definitely document that. Done.